packages/compass-components/src/components/index-keys.spec.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { render, cleanup, screen, within } from '@testing-library/react';
+import { expect } from 'chai';
+import type { IndexDirection } from 'mongodb';
```

Comment:
AFAIU the idea of compass-components is not to know about driver, mongodb, server, etc explicitly. This is also not used anywhere outside of compass-indexes package. Is there any particular reason we are putting this in compass-components?

/cc @Anemy any thoughts?

Reply:
It makes more sense to me to move this inside the index plugin unless we are already using it somewhere else, not sure if we have similar code already, in which case, we could probably first see if we can decompose this component in parts that are a bit more flexible to reuse.

Reply:
Yeah, this sounds about right to me. At the very least I would really tried to avoid having a driver as a dependency here: there is not reason for component interface to be directly attached to the driver implementation

Reply:
I'd agree let's avoid the driver implementation specifics here and the driver import if possible. As a rule of thumb I think unless we know something will be used in more packages than one we probably don't want it in Compass components. I'm thinking for this the index badge might also be used in things like explain output. That being said this component isn't much more than a wrapper around a badge with an icon and text so I think it would be find to build it in both places with those building blocks.

Reply:
This component is now being used in 2 places: here (in indexes) and aggregation explain. it will also be used in the Explain tab as well. Not certain if it makes sense to have it in three different places. 
I kept it in compass-components while removing the dependency. Is there a better way to handle this?

Reply:
This is just a thought. If this component is reused then there is nothing wrong with keeping the code in a shared place like this. However, when we decide to generalize a solution, we may want to try to reframe it in a way that lets us share what is actually reused and that won't change with context.

I'm not suggesting doing anything specific in this case, but just to give an example: I would consider sharing just sharing a single `IndexKey` component instead of a "list" as is pretty straightforward to build the list where is used and the layout may depend on the context where is used. 

By the same token, there is probably not much that is special about the badge, so .. again is just an example, we could consider only having an "IndexTypeIcon". 

>  it will also be used in the Explain tab

I really hope that that won't be a 3rd place cause all the code for explains should be in the same package eventually.